src/libraries/System.Private.DataContractSerialization/src/System/Xml/XmlBaseReader.cs

```diff
@@ -902,7 +911,7 @@ public override XmlNameTable NameTable
             {
                 if (_nameTable == null)
                 {
-                    _nameTable = new NameTable();
+                    _nameTable = new QuotaNameTable(this, _quotas.MaxNameTableCharCount);
                     _nameTable.Add(xml);
```

Comment:
We might need to add this to breaking changes, that we now honor the MaxNameTableCharCount